fullstack/js/price-calculator/app/managers/base-price-manager.js

```diff
@@ -1,25 +1,36 @@
-import ProductBasePrice from '../models/base-price'
+import ProductBasePrice from '../models/base-price';
 
 export default class BasePriceManager {
-  data = {}
+  data = {};
 
   constructor(jsonFile) {
-    this.data = this.readJson(jsonFile)
+    this.data = this.readJson(jsonFile);
   }
 
   readJson(jsonFile) {
-    let data = {}
-    const rawData = require(jsonFile)
+    let data = new Object();
+    const rawData = require(jsonFile);
 
-    for (const rawItem in rawData) {
-      const productBasePrice = new ProductBasePrice(rawItem)
-      data[productBasePrice.productType] = productBasePrice
+    for (const rawItem of rawData) {
+      const productBasePrice = new ProductBasePrice(
+        rawItem['product-type'],
+        rawItem.options,
+        rawItem['base-price']
+      );
+
+      if (data[productBasePrice.productType]) {
+        data[productBasePrice.productType].basePrices.push(productBasePrice);
+      } else {
+        data[productBasePrice.productType] = {};
+        data[productBasePrice.productType].basePrices = [];
+        data[productBasePrice.productType].basePrices = [productBasePrice];
+      }
```

Comment:
Should we have a single if to initialise the basePrices array here?

Reply:
I left it like this on purpose and is documented here https://redbubble.atlassian.net/wiki/spaces/ENG2/pages/2819293193/JS+Price+Calculator+Guidebook#%F0%9F%A4%93-Known-issues